idl/internal/parser.go

```diff
@@ -27,13 +27,13 @@ func init() {
 }
 
 // Parse parses the given Thrift document.
-func Parse(s []byte) (*ast.Program, []ParseError) {
+func Parse(s []byte) (*ast.Program, NodePositions, []ParseError) {
```

Comment:
I don't believe the package is consider "public API", but we could also think about returning a struct value here to avoid ballooning the number of return values over time.